### ISSUE TITLE
AI Reference Panel Meeting No. 3 - October 2021

### ISSUE THREAD

**Issue by sarahcleary** [APSIMInitiative/ReferencePanel#97]
The third AI RP meeting is planned for Wednesday 13/10.
If you have agenda items for this meeting, can you please add them (as you think of them) to this issue?
thanks.
@APSIMInitiative/reference-panel; @uqschap7; @erik-van-oosterom
@APSIMInitiative/steering-committee

**Comment by sarahcleary** [comment]
Proposed Agenda Items

1. APSIM virtual issue - @kchenu 
2. Update on Workshops/Sprints - any of interest/for update?
3. APSIM Training
- Online Seminar Series
- ASA in Feb
- APSIM Week
- Plans for 2022
4. Plausibility Testing (continuing discussion at last AI SC Meeting)
5. AI Software Position 

Any other suggestions?

**Comment by sarahcleary** [comment]
Review/comparison of content of revised RP processes/meeting structure -  @jbrider and @sarahcleary

**Comment by uqschap7** [comment]
Can you put all this in the calendar invite, i.e. agenda etc…

**Comment by sarahcleary** [comment]
@uqschap7 - this current thread is to gather agenda items from all interested parties.  Once I have input, I will put together the agenda and put a link in the Agenda for ease of access.  Having it on GitHub allows people to see what we might be discussing and to indicate interest in attending.   I also utilise this issue for the Minutes so everyone can access these.  Does that make sense?

**Comment by sarahcleary** [comment]
Please find attached a draft agenda for your input.  
 
| Time       | Topic     | Lead     |
| :------------- | :---------- | -----------: |
| 8:50 | Welcome   | Jason Brider @jbrider     |
| 9:00   |<p>RP Software Developments and Processes</p><ul><li>Workshop and/or Software Development Updates </li> <ol><li>Update from @sarchontoulis on Root Modelling Work https://github.com/APSIMInitiative/ApsimX/issues/6725 </li> <li>Update from @Keith-Pembleton on Soils Workshop/project - https://github.com/APSIMInitiative/ReferencePanel/issues/18</li> <li> Update from @hol353 on [Improved systems for sensibility tests for all crop models](https://github.com/APSIMInitiative/ApsimX/issues/6517)</li></ol><li>Plausibility Testing - as per discussion at AI SC meeting - potentially postpone until after Sorghum release?</li></ol><li>AI Software Position @jbrider</li></ol><li>Review and discussion on how online GitHub 'weekly' review process is going by @APSIMInitiative/reference-panel</li><ol><li>Comparison of monthly review meetings and 2021 Structure </li></ol><li>Online Review process</li> <ol><li>Comments on Improving Processes</li> <li>Models to be reviewed</li></ol><li> Development Plan <ul><li>[Autodoc failures](https://github.com/APSIMInitiative/ApsimX/issues/6409) led by @HamishBrownPFR </li><li> APSoil update from @hol353 </li><li> New developments? </li><li> Projects for possible funding? https://github.com/APSIMInitiative/ReferencePanel/issues/62 | @APSIMInitiative/Reference-Panel; @jbrider; @hol353; @Keith-Pembleton  |
| 10:20 | Break| |
| 10:30 | APSIM Training<ul><li>Online Seminar Series Review of current list videos - https://github.com/APSIMInitiative/ReferencePanel/issues/87<li>Review of interest via GitHub, Distribution List, Website, RG etc.. </li><li> Plans for 2022</li><li> ASA in February - Requirements, Promotion</li><li> APSIM Week</li><li> APSIM-related teaching activities at Iowa State</li><li>Update on APSIM special issue by @kchenu|@APSIMInitiative/Reference-Panel; @kchenu; @femiguez   |
| 11:20 |AI RP Chair for 2022|Jason Brider @jbrider  |
| 11:30 |Close| |

Please let me know if you wish to invite others that can contribute to these discussions.  

Please ensure you have reviewed GitHub recently and provide any feedback on relevant issues.

**Comment by sarahcleary** [comment]
@APSIMInitiative/reference-panel - if anyone has ideas of new 'initiatives' that the AI can undertake or that the AI can assist with, please let @sarahcleary know.

**Comment by femiguez** [comment]
@sarahcleary I can provide a 5 minute update on APSIM-related teaching activities at Iowa State

**Comment by sarahcleary** [comment]
thanks @femiguez - I have added it to the agenda

**Comment by sarahcleary** [comment]
Apologies received from @EnliWang

**Comment by sarahcleary** [comment]
Apologies received from @LouisAK

**Comment by sno036** [comment]
@sarahcleary  - apologies from @sno036 as well.

**Comment by sarahcleary** [comment]
Thanks @sno036.  @MarkLieffering - are you available?

**Comment by MarkLieffering** [comment]
I guess I had better be 😊. Yep – no prob

**Comment by sarahcleary** [comment]
Apologies from @hol353.

**Comment by uqschap7** [comment]
Sounds like we should cancel this meeting?

**Comment by sarahcleary** [comment]
Apologies @uqschap7.
Yes, we will be proceeding with the meeting tomorrow as we have a quorum
DAF - @jbrider 
UQ - @peter-devoil & @kchenu 
CSIRO - @JulianneLilley 
USQ - @Keith-Pembleton 
PFR - @rcichota 
ISU - @sarchontoulis & @femiguez 
AgR - @MarkLieffering 

If any of the above is unable to attend, please let me know.

**Comment by sarahcleary** [comment]
Time       | Topic     | Lead     |
| :------------- | :---------- | -----------: |
| 8:50 | Welcome   | Jason Brider @jbrider     |
| 9:00   |<p>RP Software Developments and Processes</p><ul><li>Workshop and/or Software Development Updates </li> <ol><li>Update from @sarchontoulis on Root Modelling Work https://github.com/APSIMInitiative/ApsimX/issues/6725 </li> <li>Update from @Keith-Pembleton on Soils Workshop/project - https://github.com/APSIMInitiative/ReferencePanel/issues/18</li> <li> Update from @hol353 on [Improved systems for sensibility tests for all crop models](https://github.com/APSIMInitiative/ApsimX/issues/6517) [Postpone till next meeting]</li></ol><li>Plausibility Testing - as per discussion at AI SC meeting - [postpone until after Sorghum release]</li></ol><li>AI Software Position @jbrider</li></ol><li>Review and discussion on how online GitHub 'weekly' review process is going by @APSIMInitiative/reference-panel</li><ol><li>Comparison of monthly review meetings and 2021 Structure </li></ol><li>Online Review process</li> <ol><li>Comments on Improving Processes</li> <li>Models to be reviewed</li></ol><li> Development Plan <ul><li>[Autodoc failures](https://github.com/APSIMInitiative/ApsimX/issues/6409) led by @HamishBrownPFR </li><li> APSoil update from @hol353 </li><li> New developments? </li><li> Projects for possible funding? https://github.com/APSIMInitiative/ReferencePanel/issues/62 | @APSIMInitiative/Reference-Panel; @jbrider; @hol353; @Keith-Pembleton  |
| 10:00 | APSIM Training<ul><li>Online Seminar Series Review of current list videos - https://github.com/APSIMInitiative/ReferencePanel/issues/87<li>Review of interest via GitHub, Distribution List, Website, RG etc.. </li><li> Plans for 2022</li><li> ASA in February - Requirements, Promotion</li><li> APSIM Week</li><li> APSIM-related teaching activities at Iowa State</li><li>Update on APSIM special issue by @kchenu|@APSIMInitiative/Reference-Panel; @kchenu; @femiguez   |
| 10:50 | Break| |
| 11:00 |Discussion on uptake of Next Gen |Peter Thorburn @PeterThorburn |
| 11:30 |AI RP Chair for 2022|Jason Brider @jbrider  |
| 11:40 |Close| |

**Comment by sarahcleary** [comment]
@APSIMInitiative/reference-panel - just letting you know that we've covered the majority of the agenda. 
Now on a break.  Please phone back in prior to 11 (BNE time) if joining the Discussion on uptake of Next Gen.

**Comment by PeterThorburn** [comment]
I’m in workshop until 11 am. So I won’t be able to join early. If the meeting ends early, that’s fine. There will be other opportunities to have that conversation.

PT

_______________________________________________________________________
Dr Peter Thorburn | Research Group Leader: Systems Program | CSIRO Agriculture & Food
    Ph (0)7 3214 2412 | Mobile: +61 (0)417 073 173 | Fax: +61 (0)7 3214 2463

**Comment by jbrider** [comment]
@APSIMInitiative/reference-panel 
https://github.com/APSIMInitiative/ApsimX/issues/6870
After discussion with @hol353 We propose that the easiest solution for a 'training version' was to enable people to create a url link to a specific release - which could be published as part of a course's requirements.
We could also enable searching by release number on the downloads page, and also enable similar functionality to the Upgrade process.

**Comment by sarahcleary** [comment]
@sarchontoulis - are you comfortable with the above solution for the training version  - https://github.com/APSIMInitiative/ReferencePanel/issues/97#issuecomment-945277907

**Comment by sarchontoulis** [comment]
Yes, sounds like a good solution!

**Comment by sarahcleary** [comment]
# 2021-10 Minutes
### Members: Video/Teleconference:
@jbrider; @peter-devoil; @kchenu; @JulianneLilley; @Keith-Pembleton; @rcichota; @sarchontoulis; @femiguez; @MarkLieffering; @uqschap7; @erik-van-oosterom
### October 13th 2021 - 8:50 am AEST
## Agenda Item 

### Welcome
Welcome from @jbrider to new members  and @femiguez and @rcichota
@jbrider provided a summary of the current structure of the RP meetings and expectations of RP members to review GitHub regularly. Note that we are still working under the requirement that there is need for agreement from AI members for all decisions are being made. Each organisation has one vote.  
### RP Software Developments and Processes
## Workshop and/or Software Development Updates 
Update from @sarchontoulis on [Root Modelling Work](https://github.com/APSIMInitiative/ApsimX/issues/6725)
Not much to report as busy with other projects.   
Major root data has been uploaded into APSIM GitHub and working on Maize and Soybean model.
Now waiting for some improvements in the models prior to be able to develop improvements in the root equations.  Interelated work. Have met with both @HamishBrown and @jbrider

**Update from @Keith-Pembleton on Soils Workshop/project** - https://github.com/APSIMInitiative/ReferencePanel/issues/18
No progress due to capacity issues. 

Update from @hol353 on [Improved systems for sensibility tests for all crop models](https://github.com/APSIMInitiative/ApsimX/issues/6517) 
Postpone update till next meeting

**Plausibility Testing** - as per discussion at AI SC meeting
Postpone until after Sorghum release.  Issues should be addressed in this process.

**AI Software Position** 
FYI – Graeme has requested to keep the existing arrangement.  So @hol430 we continue with 50%UQ ; 50% AI.

**Review and discussion on how online GitHub 'weekly' review process is going - Comparison of monthly review meetings and 2021 Structure** 

Main difference is that there is little conversation about software developments arising from new issues.  In the previous monthly meetings there was ‘incidental conversations’ but little if no interaction via GitHub that would consider to be a replacement.  Maybe occurring in these longer meetings but not via GitHub.
Some of these discussions are covered as part of the online review process. 

**Models for Review**
Sorghum Review – delayed due to @jbrider additional role.  

**Development Plan**
[Autodoc failures](https://github.com/APSIMInitiative/ApsimX/issues/6409) – noted that @hol430 submitted a large pull request yesterday so assume progress is being made.  

**APSoil update**
@sarahcleary provided an updated.
Noted that CSIRO agreed that the AI owns APSoil, but recommended better management of the data and ensure that when data is added, they are noted under what conditions they are adding the data to APSoil.  Improvement on communication which is tied to the revamp the APSIM download pages and process.

**New developments** 
@Keith-Pembleton provided an update on a new PhD student modelling perennial grain in Next Gen.  Once started, will enter into the Development Plan.  Working with DPI (NSW) who hold a lot of the data.  Utilising the Lucerne model as the basis.  

## Projects for possible funding 
Noted the potential funding opportunity for [Improved systems for sensibility tests for all crop models](https://github.com/APSIMInitiative/ApsimX/issues/6517) – noting lack of capacity of @hol353
Also, note @sno036 ’s comment. 
Please add comment if there are any thoughts.

## APSIM Training
## Online Seminar Series
Review of current list videos - https://github.com/APSIMInitiative/ReferencePanel/issues/87<li>
Note there has been a break in videos due to lack of capacity from Adam and Val. 
Will circulate new videos as they are recorded.

## Plans for 2022
Even if March 2022 ends up possible, but too close to rescheduled ASA 2021
Needs to be face to face.
Propose second half of 2022 – possibly September 2022
Not publicizing any potential date
Maybe need to book rooms at QBP in December to secure rooms, including the lecture theatre.
Review teaching break/school holidays etc..  

## Training for Next Gen
ASPIM Week was originally part of the solution
What can we provide in the interim?
ISU intending on converting their training materials to Next Gen in January for the upcoming workshop mid-year.  Noting there will be some specific examples for the US – corn and soyabean.
Same with USQ (@keith-Pembleton) – to convert materials from Classic to Next Gen -  but relies on wheat, chickpea and sorghum, so waiting on sorghum.  Not training in modelling but do teach mechanics of setting up a simulation. 
@sarchontoulis raised a concern about having both Maize Model and SWIM in release.  But also, to ensure there is a single version for this training to avoid training materials to become obsolete and require changing close to the workshop.
Discussion about the possibility and value of converting the training materials from APSIM Classic to APSIM Next Gen.  Existing materials would be of value, but no capacity within the group.  

## APSIM Special Issue
@kchenu provided an update.   26 EOIs have been received. 4 Submissions, 1being reviewed.
Expect more to come.  Diverse submissions.  Ongoing submissions, Deadline of April 2022.  

## ISU Update
@femiguez provided a summary of his graduate course in crop and soil modelling.  Offered to adapt one of his lectures to be used as an APSIM Online Seminar Video.  He has requested some suggestions of topics.  Uses both APSIM Classic and Next Gen.  In 2021, using Next Gen in this course.  Does create examples as part of this course and happy to take requests for examples that could be utilised more broadly.

## Discussion on Stability of APSIM Next Gen for teaching
APSIM Classic ‘stable’ as it isn’t being developed so teaching materials don’t need modification
@keith-pembleton commented that online students don’t cope well when something doesn’t work – so if the instructions need to be modified due at change in the model or the results are not identical as the example.
@femiguez ensures the students download a version and work with that during the semester to avoid changes/errors.  
Version change can cause issues with both teaching and project work.
Increased issue due to the frequency of the updates.  
Discussion on an ‘educational’ version of APSIM.  Proposed an option of Next Gen ‘teaching’ version. 
Key action is to ensure bugs are reported so they can be fixed
**ACTION: **  @hol353  on providing a 'Training Version YYYY' near the end of the year (so materials may be updated for the following year.  These could then be easily distinguished amongst a list of other versions.

## ASA 2021
As per the AI SC Meeting, should we look at any APSIM training at the ASA conference.
**ACTION: **  @sarahcleary to prepare material for the ASA conference

## AI RP Chair for 2022
In general, the chair rotates through the AI members. 
@sarchontoulis  would be happy to Chair but in 2023 due to workload
@HamishBrownPFR  – has been asked, but is checking internally.  He is currently on leave
**ACTION:**  Await @HamishBrownPFR a'snswer


## Discussion on uptake of Next Gen led by @PeterThorburn
Developers only working on APSIM Next Gen
Users not migrating.
Has the RP thought about either informally or formally approaching the APSIM community to find out why users are not migrating.
@PeterThorburn hasn’t moved to Next Gen predominately due to versioning.  
What are barriers? 
Potential funding from the AI SC if there is ‘evidence’ of the barriers which need to be removed to help users migrate.
@Keith-Pembleton – his team was due to lack of crop models.
Possibly tell who is actually using APSIM vs downloading.
@rcichota  – versioning issue – if one person in the group changes version, then others can’t utilise which causes issues for the project.
@peter-devoil – functionality

Surveys – what are we trying to achieve?  What questions to ask.

Proposal from @PeterThorburn – everyone talk to 3-5 people – do you use Next Gen, and if not, why?
And/or  Survey on something like RG – survey with say, 4 questions.  E.g. SWIM
@sarchontoulis  – time.  Need to invest time to come up to speed with Next Gen.  If APSIM Classic doesn’t exist, would push everyone to Next Gen.  Interface has changed, looks different, terms are different
Note that APSIM Classic might stop being able to be utilised to the general user when a Microsoft upgrade occurs.  Need to get this communication out there.
@femiguez – functionality;  confusing versioning for normal users – or official release every 6 months might help to solve the problem
Would be helpful to have ‘cheat sheets’ which showed the difference with APSIM Classic and Next Gen, with reference to the terms used.
**ACTION:**  Identify key people to talk to.
@Keith-Pembleton  to go through last 12 months of APSIM Classic papers and produce an email contact list  

@jbrider  updated the RP on a discussion with the AI SC.  Noted that there was discussion on the ideal situation of adding resources for bug fixing etc..  Training people to really understanding the system.  AI SC might be interested to partially fund support with another AI member.   50:50 model.
@Keith-Pembleton  should have someone (approx. 1 month of funding) for implementation into APSIM.
Current is more around capacity.
**ACTION:**  Keith-Pembleton to draft a letter for discussion at the AI SC.
